config/routes.rb

```diff
@@ -14,7 +14,11 @@
     resources :items, only: [:new, :create, :edit, :update, :destroy]
   end
 
-  resources :shops, only: [:index]
+  resources :shops, only: [:index, :show]
+
+  namespace :shop do
```

Comment:
Made all the slug changes, and am getting the error "No route matches {:action=>"show", :controller=>"shops", :shop=>"shop-1"} missing required keys: [:id]"

Example from class has the route
namespace :stores, path: ":store_name", as: :store do
....
end

but I also tried that and still get the same error. It is still expecting an id even thought we want it to expect a slug, but I'm not sure where the code that makes it expect that is.